internal/bminventory/inventory.go

```diff
@@ -2103,60 +2094,65 @@ func (b *bareMetalInventory) updateHostRoles(ctx context.Context, params install
 
 func (b *bareMetalInventory) updateHostNames(ctx context.Context, params installer.UpdateClusterParams, db *gorm.DB, log logrus.FieldLogger) error {
 	for i := range params.ClusterUpdateParams.HostsNames {
-		log.Infof("Update host %s to request hostname %s", params.ClusterUpdateParams.HostsNames[i].ID,
-			params.ClusterUpdateParams.HostsNames[i].Hostname)
-		var host models.Host
-		err := db.First(&host, "id = ? and cluster_id = ?",
-			params.ClusterUpdateParams.HostsNames[i].ID, params.ClusterID).Error
+		hostName := params.ClusterUpdateParams.HostsNames[i]
+		log.Infof("Update host %s to request hostname %s", hostName.ID, hostName.Hostname)
+		var host, err = getHost(hostName.ID.String(), params.ClusterID.String(), db, log)
 		if err != nil {
-			log.WithError(err).Errorf("failed to find host <%s> in cluster <%s>",
-				params.ClusterUpdateParams.HostsNames[i].ID, params.ClusterID)
 			return common.NewApiError(http.StatusNotFound, err)
 		}
-		if err = hostutil.ValidateHostname(params.ClusterUpdateParams.HostsNames[i].Hostname); err != nil {
+		if err = hostutil.ValidateHostname(hostName.Hostname); err != nil {
 			log.WithError(err).Errorf("invalid hostname format: %s", err)
 			return err
 		}
-		err = b.hostApi.UpdateHostname(ctx, &host, params.ClusterUpdateParams.HostsNames[i].Hostname, db)
+		err = b.hostApi.UpdateHostname(ctx, host, hostName.Hostname, db)
 		if err != nil {
 			log.WithError(err).Errorf("failed to set hostname <%s> host <%s> in cluster <%s>",
-				params.ClusterUpdateParams.HostsNames[i].Hostname, params.ClusterUpdateParams.HostsNames[i].ID,
+				hostName.Hostname, hostName.ID,
 				params.ClusterID)
 			return common.NewApiError(http.StatusConflict, err)
 		}
 	}
 	return nil
 }
 
+func getHost(hostId string, clusterId string, db *gorm.DB, log logrus.FieldLogger) (*models.Host, error){
```

Comment:
@YuviGold didn't you already create a similar utility function? Or was that just for clusters?

Reply:
Right. Please use `common.GetHostFromDB`
https://github.com/openshift/assisted-service/blob/927f8bbdf395e2ee830749eb672283d1089396b8/internal/common/db.go#L135-L142

Reply:
It's a different Host type: common.Host instead of models.Host

Reply:
> It's a different Host type: common.Host instead of models.Host

which is just a models.Host with an extra field... @rollandf what is this weird distinction between `common.Host` and `models.Host`? why isn't the Approved field in the `swagger.yaml`?

Reply:
It is not in swagger because it is not exposed in the API.
If you look at the Cluster, you can see that there also additional fields that are not in swagger.
The field is in the DB, look in #1183 for how it was added.

Reply:
```
	return db.AutoMigrate(&models.Host{}, &Cluster{}, &Event{}).Error
	return db.AutoMigrate(&Host{}, &Cluster{}, &Event{}).Error
```

Reply:
look like i can't use this utility method... 
may i resolve this conversation?

Reply:
@michaellevy101 I don't see a reason why we need another function. The common module and functions I made were to avoid this kind of confusion.

Reply:
All the code uses the models.Host and not the common.Host and we fetch the host from the DB 5 times (I think).
I just extracted this code to a new method to be more readable but not changed the logic...
I agree that we should use your new method but IMO this PR is not the right place, let's do this in a dedicated PR

Reply:
I agree with Yuval's comment. If there is a known utility use it and then pass on just the models.Host portion. I can see getHost() is used only once in this specific commit

Reply:
@slaviered  please see the whole file...

Reply:
It was decided to open a new task [MGMT-4801](https://issues.redhat.com/browse/MGMT-4801) to make this refactor aligned throughout the server.